### PR TITLE
Replace Dhall with TOML using toml-reader

### PR DESCRIPTION
**Pull request by tfausak**
Same idea as #118 but using `toml-reader` instead of `tomland`. See also #117.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,8 +55,8 @@ stack build
 
 ## Calling Weeder
 
-To call Weeder, you first need to provide a configuration file, `weeder.dhall`. Weeder uses
-[Dhall](https://dhall-lang.org) as its configuration format, and configuration
+To call Weeder, you first need to provide a configuration file, `weeder.toml`. Weeder uses
+[TOML](https://toml.io/en/) as its configuration format, and configuration
```
Comment (ocharles) [inline README.md:59]: I think we should remove "and configuration files have the type...", as that's fairly Dhall specific.

File: src/Weeder/Config.hs [hunk 2]
```diff
@@ -3,14 +3,13 @@
...
-module Weeder.Config ( Config(..), config ) where
+module Weeder.Config ( Config(..) ) where
 
 -- containers
 import Data.Set ( Set )
-import qualified Data.Set as Set
 
--- dhall
-import qualified Dhall
+-- toml
```
Comment (ocharles) [inline src/Weeder/Config.hs:11]:
```suggestion
-- toml-reader
```

I think? I should release the tool we use at CircuitHub to do this automatically!

File: src/Weeder/Main.hs [hunk 3]
```diff
@@ -24,8 +25,8 @@ import qualified Data.Set as Set
 -- text
 import qualified Data.Text as T
 
--- dhall
-import qualified Dhall
+-- toml
```
Comment (ocharles) [inline src/Weeder/Main.hs:28]:
```suggestion
-- toml-reader
```

**Review by ocharles — APPROVED** [review]
LGTM! Thanks!

**Comment by tfausak** [comment]
I'm not sure why the test suite is failing, but I don't think it's related to my changes.

**Comment by ocharles** [comment]
It's not, looks like I need to update CI. I'll sort that.

**Comment by tfausak** [comment]
Perhaps the changes from #115 would be useful?

**Comment by ocharles** [comment]
@tfausak Ok, I think that should get it. I've taken the GHC 9.6 changes out from this PR for now as they aren't verified in CI. I'll get on that next!

**Comment by tfausak** [comment]
Sounds good to me. Thanks!

**Pull request merged**